### PR TITLE
A bit of tuning and optimisation

### PR DESCRIPTION
**Pull request by config-i1**
Things I changed:
1. Get loss, from adam(), don't allow changing it. Otherwise we are facing weird stuff with adam() estimated using whatever and then LASSO/RIDGE based on it.
2. Renamed x0 into lambda for consistency with adam()
3. Import functions that are used inside, don't do "::" thingy
4. Simplified some checks
5. Added ci and co parameters for the ro() function
6. Added ellipsis (...) for a better control of adam() by user
7. Simplified nloptr call

I hope this is helpful :).

### CODE REVIEW

**Review by nikdraka — APPROVED** [review]

File: R/estimate_lambda.R [hunk 1]
```diff
@@ -24,11 +24,11 @@
 #' library(smooth)
 #' library(greybox)
 #'
-#' # Estimate an exponential smoothing model first
-#' fit <- adam(AirPassengers, loss = "MSE")
+#' # Estimate an exponential smoothing model with RIDGE first
+#' fit <- adam(AirPassengers, loss = "RIDGE", lambda = 0.5)
```
Comment (nikdraka) [inline R/estimate_lambda.R]: But what if a user wants "LASSO" instead of "RIDGE"?

Comment (nikdraka) [reply R/estimate_lambda.R]: Oh sorry! this is just an example! I thought this was a part of the function :)

**Review by nikdraka — APPROVED** [review]
